pyproject.toml

```diff
@@ -319,6 +319,8 @@ all = [
 isort = ">=5.10.1"
 black = ">=22.7.0"
 ruff = "0.0.286"
+# Type checking only works with the pydantic.v1 compat module from pydantic v2
+pydantic = "^2"
```

Comment:
Is this just because we only type check against the locked dependencies? It looks like it would be fine to run it regardless.

Reply:
/resolve. I answered in https://github.com/matrix-org/synapse/pull/16332#discussion_r1329133969.